### PR TITLE
Update max sentence length to 84

### PR DESCRIPTION
**Pull request by GreenzieNick**
### Purpose of PR:
This PR increases the Max allowed sentence length of NMEA messages accepted by the NTRIP client from 83 to 84. This is necessary because when we use the FixPosition, we use the "High-Precision" NMEA output format. This does the following according to FixPosition Integration Manual:
"High-Precision: Enable output of non-standard data fields with more precision."
This means longer sentence lengths which allows for more precision which necessitates the need for the increase in max sentence length.
This was not seen as an issue as 84 byte messages were not commonly seen until bringing up the NOVA mowers in the Cincinnati area where we started seeing many 84 byte messages.

### CODE REVIEW

**Review by rm-greenzie — APPROVED** [review]

**Comment by edavies64** [comment]
@GreenzieNick is 84 bytes the max size we're expecting with the `"High-Precision" NMEA output format` or should we go even larger?
Alternatively, why wouldn't we adjust the FixPosition's config to use Strict NMEA instead?

**Comment by GreenzieNick** [comment]
> @GreenzieNick is 84 bytes the max size we're expecting with the `"High-Precision" NMEA output format` or should we go even larger? Alternatively, why wouldn't we adjust the FixPosition's config to use Strict NMEA instead?

For the max size expected, I am not sure. I haven't been able to find a maximum length possible in my searching. the largest I have seen is 84 bytes.
We could conform to standard NMEA messages. We would just have to reconfigure all FixPositions when possible for consistency. We would also lose precision as the number of decimal places in the location portion of the messages will be truncated to fit the standard message. I do not know how this would affect our system. I would want a comparison done of how much of a difference in precision there is between standard and high-precision messages

**Comment by edavies64** [comment]
> > @GreenzieNick is 84 bytes the max size we're expecting with the `"High-Precision" NMEA output format` or should we go even larger? Alternatively, why wouldn't we adjust the FixPosition's config to use Strict NMEA instead?
> 
> For the max size expected, I am not sure. I haven't been able to find a maximum length possible in my searching. the largest I have seen is 84 bytes. We could conform to standard NMEA messages. We would just have to reconfigure all FixPositions when possible for consistency. We would also lose precision as the number of decimal places in the location portion of the messages will be truncated to fit the standard message. I do not know how this would affect our system. I would want a comparison done of how much of a difference in precision there is between standard and high-precision messages

Can you reach out of FP to see what the max size is? And/or what the tradeoffs would be using standard NMEA vs high precision.

**Comment by GreenzieNick** [comment]
> > > @GreenzieNick is 84 bytes the max size we're expecting with the `"High-Precision" NMEA output format` or should we go even larger? Alternatively, why wouldn't we adjust the FixPosition's config to use Strict NMEA instead?
> > 
> > 
> > For the max size expected, I am not sure. I haven't been able to find a maximum length possible in my searching. the largest I have seen is 84 bytes. We could conform to standard NMEA messages. We would just have to reconfigure all FixPositions when possible for consistency. We would also lose precision as the number of decimal places in the location portion of the messages will be truncated to fit the standard message. I do not know how this would affect our system. I would want a comparison done of how much of a difference in precision there is between standard and high-precision messages
> 
> Can you reach out of FP to see what the max size is? And/or what the tradeoffs would be using standard NMEA vs high precision.

Will still ask about the size, but I found this comparison table in their manual about the differences in the formats from a data accuracy stand point.
image

**Comment by seebq** [comment]
Hey @GreenzieNick I fixed your build by adding a specific flag to the debian/rules.  I'm not sure what the ramifications are, but it definitely disables to option that was causing the build to fail.

File: .github/workflows/main_workflow.yml [hunk 1]
```diff
@@ -25,9 +25,7 @@ jobs:
   debianize:
     name: "Debianization"
     timeout-minutes: 30
-    runs-on:
-      - self-hosted
-      - greenzie
+    runs-on: ubuntu-latest
```
Comment (seebq) [inline .github/workflows/main_workflow.yml]: You can probably switch this back.

Comment (GreenzieNick) [reply .github/workflows/main_workflow.yml]: I have switched it back

**Comment by edavies64** [comment]
> Hey @GreenzieNick I fixed your build by adding a specific flag to the debian/rules. I'm not sure what the ramifications are, but it definitely disables to option that was causing the build to fail.

Awesome. Nice black magic fix!

File: launch/ntrip_client.launch [hunk 2]
```diff
@@ -60,7 +60,7 @@
     <param name="rtcm_frame_id" value="odom" />
 
     <!-- Optional parameters that will allow for longer or shorter NMEA messages. Standard max length for NMEA is 82 -->
-    <param name="nmea_max_length" value="83" />
+    <param name="nmea_max_length" value="84" />
```
Comment (edavies64) [inline launch/ntrip_client.launch:66]: Please add in a comment here about why we're doing this. And the max byte sizes for both.
NMEA --> 82
High precision NMEA --> 84(?)

Comment (GreenzieNick) [reply launch/ntrip_client.launch:66]: I have added a comment

**Pull request merged**